### PR TITLE
[JBPM-10015] Allow to set suspendUntil during user task suspend operation as parameter.

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-10015 
depends on 
https://github.com/kiegroup/droolsjbpm-knowledge/pull/589

### CODE REVIEW

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/SuspendTaskCommand.java [hunk 1]
```diff
@@ -33,21 +35,23 @@
 public class SuspendTaskCommand extends UserGroupCallbackTaskCommand<Void> {
 	
 	private static final long serialVersionUID = 5486559063221608125L;
+    private Map<String, Object> parameters;
```
Comment (gmunozfe) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/SuspendTaskCommand.java:38]: Add @XmlTransient annotation

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/WorkflowBridgeTaskLifeCycleEventListener.java [hunk 2]
```diff
@@ -39,20 +39,23 @@ public WorkflowBridgeTaskLifeCycleEventListener(RuntimeManager runtimeManager) {
 
     @Override
     public void afterTaskSuspendedEvent(TaskEvent event) {
-        executeWork(event.getTask(), HumanTaskNodeInstance.SUSPEND_SIGNAL);
+        String suspendUntilExpression = (String) event.getMetadata().get("suspendUntil");
```
Comment (gmunozfe) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/WorkflowBridgeTaskLifeCycleEventListener.java:42]: Perhaps it's more clear to define a constant for "suspendUntil" as it's used several times in this class.

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/task/SynchronizedTaskService.java [hunk 3]
```diff
@@ -543,6 +543,13 @@ public void suspend(long taskId, String userId) {
         }
     }
```
Comment (gmunozfe) [inline jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/task/SynchronizedTaskService.java:544]: probably the method with 2 params is never called now

**Review by gmunozfe — APPROVED** [review]
Looks good to me, just a few minor comments, great work @elguardian !

File: jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/UserTaskService.java [hunk 4]
```diff
@@ -343,6 +352,16 @@
      */
     void suspend(String deploymentId, Long taskId, String userId);
 
+    /**
+     * Suspend a claimed task.
+     * 
+     * @param deploymentId
+     * @param taskId
+     * @param userId
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-services/jbpm-services-api/src/main/java/org/jbpm/services/api/UserTaskService.java:360]: added

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/ProcessHumanTaskTest.java [hunk 5]
```diff
@@ -321,6 +321,45 @@ public void afterTaskResumedEvent(TaskEvent event) {
...
+        assertProcessInstanceActive(processInstance.getId(), ksession);
+
+        // let john execute Task 1
+        List<TaskSummary> list = taskService.getTasksAssignedAsPotentialOwner("john", "en-UK");
+        TaskSummary task = list.get(0);
+        logger.info("John is executing task {}", task.getName());
+        taskService.start(task.getId(), "john");
+        logger.info("John is suspending task {}", task.getName());
+        long start = System.currentTimeMillis();
+        taskService.suspend(task.getId(), "john", Collections.singletonMap("suspendUntil", "PT10S"));
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-test-coverage/src/test/java/org/jbpm/test/functional/task/ProcessHumanTaskTest.java:353]: to 3 seconds

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/WorkflowBridgeTaskLifeCycleEventListener.java (hunk 2 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/WorkflowBridgeTaskLifeCycleEventListener.java:42]: create constant en HumanTaskNodeInstance

File: jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/task/SynchronizedTaskService.java (hunk 3 above)
Comment (elguardian) [reply jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/task/SynchronizedTaskService.java:544]: need to keep this for backward compatibility

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/SuspendTaskCommand.java (hunk 1 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/commands/SuspendTaskCommand.java:41]: you are right

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/TaskEventSupport.java [hunk 6]
```diff
@@ -105,9 +105,10 @@ public void fireBeforeTaskResumed(final Task task, TaskContext context) {
         }
     }
     
-    public void fireBeforeTaskSuspended(final Task task, TaskContext context) {
+    public void fireBeforeTaskSuspended(final Task task, TaskContext context, Map<String, Object> parameters) {
         if ( hasListeners() ) {
             final TaskEventImpl event = new TaskEventImpl(task, context);
+            event.getMetadata().putAll(parameters);
```
Comment (rbarriuso) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/events/TaskEventSupport.java:111]: It's been reported that this throws an exception if `parameters` is `null`

**Pull request merged**